### PR TITLE
Caddy Webserver for Frontends

### PR DESCRIPTION
**Pull request by ianliuwk1019**
Caddy webserver for frontends.

Add Caddyfile for serving frontend.
Dockerfile change to build and deploy frontend in container with Caddy.
Migrate content-security-policy settings from previous server.js (using Express) to Caddy.
Verify on browser frontends are running with no issue.
Remove server.js (public/admin) and 'helmet.js' from package.json dependency.

### CODE REVIEW

File: admin/Caddyfile [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Caddy's configuration file to serve up frontend. (Require Caddy installed/downloaded in system)
+# Local: Assume local build is ready at './dist/admin'; at root working directory run command "caddy run --watch" at console.
+#        Then access local frontend app: http://localhost:4300/admin/
+
+:4200 {
+    log {
+        level DEBUG
```
Comment (basilv) [inline admin/Caddyfile]: I'm not sure we want to keep debug logging, seems unnecessary.

File: admin/Caddyfile [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    # Good practice to compress responses.
+    encode zstd gzip
+
+    root /* ./dist/admin
+
+    header {
+        # allow caching for 5 minutes, for refreshing after deployments.
+        Cache-Control "public, max-age=300"
+        Content-Security-Policy "connect-src 'self' https://*.gov.bc.ca http://*.gov.bc.ca;form-action 'self';img-src 'self' https://server.arcgisonline.com data: https://services.arcgisonline.com data: https://maps.gov.bc.ca;frame-src 'self' https://*.gov.bc.ca;default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;frame-ancestors 'self';object-src 'none';script-src 'self';style-src 'self' https: 'unsafe-inline';upgrade-insecure-requests;"
```
Comment (basilv) [inline admin/Caddyfile:23]: You could add a comment explaining that the header definition has to remain on a single line.

File: admin/Caddyfile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+   # 'handle_path' does the match of the path and stripping the matched prefix out.
+    handle_path /admin/* {
+        try_files {path} {path}/ {file} /index.html
+    }
+
+    file_server
+}
+
+##################
```
Comment (basilv) [inline admin/Caddyfile]: Probably don't need to keep all these comments, or merge into content above. Derek can make the final call if you are unsure.

File: admin/Dockerfile [hunk 4]
```diff
@@ -10,11 +10,11 @@ RUN cd libs && \
...
     npm ci && \
     npm run build:admin && \
-    rm -rf node_modules && \
-    npm i typescript@~4.1.6 express@^4.17.1 helmet@^4.6.0
+    rm -rf node_modules
 
 # Deployment container
-FROM registry.access.redhat.com/ubi8/ubi-micro
+# FROM registry.access.redhat.com/ubi8/ubi-micro
+FROM caddy:2
```
Comment (basilv) [inline admin/Dockerfile:17]: Derek will want to try to continue to use ubi-micro as the base, but then this means installing Caddy manually, you might want to leave this up to Derek to sort out.

File: admin/Dockerfile [hunk 5]
```diff
@@ -32,14 +32,13 @@ COPY --from=builder /usr/lib64/libbrotlicommon.so.1 /usr/lib64/
...
 WORKDIR /app
-COPY --from=builder /app/admin/node_modules ./node_modules
 COPY --from=builder /app/admin/dist ./dist
 COPY --from=builder /app/admin ./admin
-COPY --from=builder /app/admin/server.js .
+COPY --from=builder /app/admin/Caddyfile .
 
 # Expose port - mostly a convention, for readability
-EXPOSE 4200
+EXPOSE 4200 2019 80 443
```
Comment (basilv) [inline admin/Dockerfile]: Not sure why other ports are exposed. I favor exposing the minimum required.

File: public/Caddyfile [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Caddy's configuration file to serve up frontend. (Require Caddy installed/downloaded in system)
+# Local: Assume local build is ready at './dist/public'; at root working directory run command "caddy run --watch" at console.
```
Comment (basilv) [inline public/Caddyfile:2]: Similar comments as for admin Caddyfile.

File: public/Dockerfile [hunk 7]
```diff
@@ -10,11 +10,11 @@ RUN cd libs && \
     cd ../public && \
     npm ci && \
```
Comment (basilv) [inline public/Dockerfile:11]: Similar comments as for admin/Dockerfile

File: admin/Dockerfile (hunk 4 above)
Comment (DerekRoberts) [reply admin/Dockerfile:17]: @basilv Thanks, agreed!  Ian and I have decided to change the base later.  For now, this is bang on.  :)

**Review by DerekRoberts — APPROVED** [review]
Good job!  Lots done.  :)

**Pull request merged**